bugbear.py

```diff
@@ -229,3 +254,13 @@ def compose_call_path(self, node):
             "to the exception.",
     type=BugBearChecker,
 )
+
+B901 = partial(
+    error,
+    message=("B901: Using ``yield`` together with ``return x``. Use ``raise "
+             "StopIteration(x)`` instead of ``return x`` if this was "
+             "intentional."),
```

Comment:
The mention of async def coroutines is a good idea, but I don't consider `return x  # noqa` to be more readable than `return x`. Readability is all this check is about!